sdk/identity/identity/src/util/safeJSONParse.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Safe JSON parse.
```

Comment:
This is called a "safe" parser but it actually seems like a very _unsafe_ parser. It can return an empty object cast to `T`, regardless of what `T` actually specifies!

Why not throw an error instead of returning `{}` in both cases below? This hides a whole class of buggy behaviors.

Or, why have this function at all? Does `parse<T>(input)` save that much compared to `JSON.parse(input) as T`?

Reply:
I like the idea of throwing errors! I’ll do that.

Reply:
I’ll remove this function and let it throw naturally.